examples/jsm/modifiers/CurveModifierGPU.js

```diff
@@ -0,0 +1,234 @@
+
+	for ( let i = 0; i < numberOfPoints; i ++ ) {
+
+		const rowOffset = Math.floor( i / TEXTURE_WIDTH );
+		const rowIndex = i % TEXTURE_WIDTH;
+
+		let pt = points[ i ];
+		setTextureValue( texture, rowIndex, pt.x, pt.y, pt.z, 0 + rowOffset + ( TEXTURE_HEIGHT * offset ) );
+		pt = frenetFrames.tangents[ i ];
+		setTextureValue( texture, rowIndex, pt.x, pt.y, pt.z, 1 + rowOffset + ( TEXTURE_HEIGHT * offset ) );
+		pt = frenetFrames.normals[ i ];
+		setTextureValue( texture, rowIndex, pt.x, pt.y, pt.z, 2 + rowOffset + ( TEXTURE_HEIGHT * offset ) );
+		pt = frenetFrames.binormals[ i ];
+		setTextureValue( texture, rowIndex, pt.x, pt.y, pt.z, 3 + rowOffset + ( TEXTURE_HEIGHT * offset ) );
+
+	}
+
+	texture.needsUpdate = true;
+
+}
+
+
+function setTextureValue( texture, index, x, y, z, o ) {
+
+	const image = texture.image;
+	const { data } = image;
+	const i = CHANNELS * TEXTURE_WIDTH * o; // Row Offset
+
+	data[ index * CHANNELS + i + 0 ] = DataUtils.toHalfFloat( x );
+	data[ index * CHANNELS + i + 1 ] = DataUtils.toHalfFloat( y );
+	data[ index * CHANNELS + i + 2 ] = DataUtils.toHalfFloat( z );
+	data[ index * CHANNELS + i + 3 ] = DataUtils.toHalfFloat( 1 );
+
+}
+
+/**
+ * Create a new set of uniforms for describing the curve modifier
+ *
+ * @param { DataTexture } Texture which holds the curve description
+ */
+export function getUniforms( splineTexture ) {
+
+	return {
+		spineTexture: splineTexture,
+		pathOffset: 0, // time of path curve
+		pathSegment: 1, // fractional length of path
+		spineOffset: 161,
+		spineLength: 400,
+		flow: 1, // int
+	};
+
+}
+
+export function modifyShader( material, uniforms ) {
+
+	const spineTexture = uniforms.spineTexture;
+
+	const pathOffset = reference( 'pathOffset', 'float', uniforms );
+	const pathSegment = reference( 'pathSegment', 'float', uniforms );
+	const spineOffset = reference( 'spineOffset', 'float', uniforms );
+	const spineLength = reference( 'spineLength', 'float', uniforms );
+	const flow = reference( 'flow', 'float', uniforms );
+
+	material.positionNode = Fn( () => {
+
+		const textureStacks = TEXTURE_HEIGHT / 4;
+
+		const worldPos = modelWorldMatrix.mul( vec4( positionLocal, 1 ) ).toVar();
+
+		const bend = flow.greaterThan( 0 ).toVar();
+		const xWeight = select( bend, 0, 1 ).toVar();
+
+		const spinePortion = select( bend, worldPos.x.add( spineOffset ).div( spineLength ), 0 );
+		const mt = spinePortion.mul( pathSegment ).add( pathOffset ).mul( textureStacks ).toVar();
+
+		mt.assign( mt.mod( textureStacks ) );
+
+		const rowOffset = mt.floor().toVar();
+
+		mt.mulAssign( TEXTURE_WIDTH );
+
+		const spinePos = textureLoad( spineTexture, ivec2( mt, rowOffset.add( 0 ) ) ).xyz;
+
+		const a = textureLoad( spineTexture, ivec2( mt, rowOffset.add( 1 ) ) ).xyz;
+		const b = textureLoad( spineTexture, ivec2( mt, rowOffset.add( 2 ) ) ).xyz;
+		const c = textureLoad( spineTexture, ivec2( mt, rowOffset.add( 3 ) ) ).xyz;
```

Comment:
Please do not use `textureLoad()` in this module otherwise there is no texture filtering. This is needed though to remove the blocky appearance which is visible right now:

<img width="604" alt="image" src="https://github.com/user-attachments/assets/a63cbc5b-50c4-4352-aeb9-01fb6b93a8ab">

Reply:
Ah, the problem is that WGSL uses the home grown filtering `tsl_biquadraticTexture()` and produces artifacts (below).
Possibly need something that interpolates along the x axis only? This doesn't appear using WebGL.

![curvemod](https://github.com/user-attachments/assets/8e288505-12ba-4e73-bf73-5be77bcd5b7c)

Reply:
on experimentation, the textureLoad coords needs clamping/wrapping to within the texture size, to avoid the above pictured issue.

I think it is hitting https://www.w3.org/TR/WGSL/#textureload 

> If the logical texel addresss is invalid, the built-in function returns one of:
> The data for some texel within bounds of the texture
> **A vector (0,0,0,0) or (0,0,0,1) of the appropriate type for non-depth textures**
> 0.0 for depth textures